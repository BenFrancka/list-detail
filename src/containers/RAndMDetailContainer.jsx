import React, { Component } from 'react';
import { fetchCharacterById } from '../services/RAndMApi';
import CharacterDetail from '../components/CharacterDetail';

export default class RAndMDetailContainer extends Component {
  state = {
    loading: true,
    character: {},
  };

  async componentDidMount() {
    const character = await fetchCharacterById(this.props.match.params.id);
    this.setState({ character, loading: false });
  }

  render() {
    const { character, loading } = this.state;
    if (loading) {
      return (
        <img
          src="https://thumbs.gfycat.com/SoftEarnestGyrfalcon-max-1mb.gif"
          alt="loading spinner"
        />
      );
    }
    return (
      <CharacterDetail
        character={
          (character.id,
          character.name,
          character.image,
          character.species,
          character.status)
        }
      />
    );
  }
}
