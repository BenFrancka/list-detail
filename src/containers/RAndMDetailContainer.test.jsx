import React from 'react';
import { render, screen } from '@testing-library/react';
import RAndMDetailContainer from './RAndMDetailContainer';

describe('RAndMContainer', () => {
  it('renders a list of Rick and Morty Characters to home page', async () => {
    render(<RAndMDetailContainer />);

    screen.getByAltText('loading spinner');
  });
});
