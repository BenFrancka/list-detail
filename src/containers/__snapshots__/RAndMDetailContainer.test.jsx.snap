// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RAndMDetailContainer renders a detail view of a single character 1`] = `
<article
  aria-label="detail-view"
>
  <img
    alt="Rick Sanchez"
    src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
  />
  <h2>
    Rick Sanchez
  </h2>
  <p>
    Alive
  </p>
  <p>
    Human
  </p>
</article>
`;
